### ISSUE TITLE
Oomph setup should include target language editing support

### ISSUE THREAD

**Issue by edwardalee** [icyphy/lingua-franca#278]
I think our Oomph setup should include editors for C/C++, Python, and Typescript. How to do that?

**Comment by edwardalee** [comment]
Clarification: It seems we have at least C/C++ support in the main Eclipse. The IDE Eclipse, however, does not. On a Mac, opening a C file invokes Xcode. Also, it does not have spaces for tabs nor line numbers turned on. Is there a way for the Oomph setup to more completely set up the IDE Eclipse?

**Comment by edwardalee** [comment]
Also, it appears that the IDE Eclipse cannot be extended with additional software. Attempting to install C/C++ support produces an error saying this Eclipse is not configured to add software. The Error Log shows this:
```
eclipse.buildId=unknown
java.version=11.0.8
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_US
Framework arguments:  -product org.eclipse.platform.ide
Command-line arguments:  -product org.eclipse.platform.ide -data /Users/eal/lingua-franca-master/ws/../runtime-EclipseXtext -dev file:/Users/eal/lingua-franca-master/ws/.metadata/.plugins/org.eclipse.pde.core/Launch Runtime Eclipse/dev.properties -os macosx -ws cocoa -arch x86_64

org.eclipse.equinox.p2.ui.sdk
Warning
Sat Feb 06 14:48:38 PST 2021
Could not locate the running profile instance. The eclipse.p2.data.area and eclipse.p2.profile properties may not be set correctly in this application's config.ini file.
```

**Comment by a-sr** [comment]
First, I hope I understood the distinction of Eclipses correctly. You refer to the Eclipse that is used to develop LF itself as IDE Eclipse and the Eclipse that is started out of this IDE Eclipse and contains the actual language support for LF as Main Eclipse.

I can add editor support for these languages by configuring Oomph to install the related features. I already did this for C/C++ (as you noticed) but I guess there was a misunderstanding since I added it to the Main Eclipse.
Hence, my questions before adding the support is which Eclipse should have it? Only the IDE Eclipse or both?

Regarding the manual installation of new features, this is not possible in the Main Eclipse because it always has to comply with target platform definition. However, in the IDE Eclipse this should be possible.
I just tested it and I was able to install C/C++ support from the Eclipse Marketplace without any errors. I double clicked a header file and Eclipse opened a dialog to ask whether I wanted to install language support for this kind of files. I confirmed and it worked.
Does the error you witnessed still appear?

**Comment by edwardalee** [comment]
No, the "IDE Eclipse" in my terminology is the one launched from the "Main Eclipse". It is the one we will "productize". That is the one that cannot be extended.  I think it should have, by default, support for all the target languages, because a developer of LF programs may want to look at the generated code.  This is particularly desirable if it is not extensible.

**Comment by a-sr** [comment]
Ah ok, I added the missing language support for Python and Typescript to the IDE Eclipse configuration ([commit](https://github.com/icyphy/lingua-franca/commit/181ef9e44904d038faba18698229190b49c29ae4)).
I was a bit confused that you reported that the C/C++ support only worked in the Main Elcipse because it was supposed to be installed to the IDE Eclipse.
Nonetheless, I tested the current version of the Oomph setup and the IDE Eclipse opened a proper editor for each of the languages. I hope this will work for you too.

**Comment by edwardalee** [comment]
What about C?  Currently, if I try to open a C file from the LF IDE (the launched, not the launcher), it opens the file in Xcode, not in Eclipse (this is after git pull, Help->Perform Setup Tasks, restart Eclipse).

Edward
